system/jhtree/jhtree.cpp

```diff
@@ -2300,7 +2304,7 @@ class CLazyKeyIndex : implements IKeyIndex, public CInterface
     virtual bool hasSpecialFileposition() const { return checkOpen().hasSpecialFileposition(); }
     virtual bool needsRowBuffer() const { return checkOpen().needsRowBuffer(); }
     virtual bool prewarmPage(offset_t offset, NodeType type) { return checkOpen().prewarmPage(offset, type); }
-
+    virtual void mergeStats(CRuntimeStatisticCollection & stats) const override { return checkOpen().mergeStats(stats); }
```

Comment:
think you should change this to:
```
    virtual void mergeStats(CRuntimeStatisticCollection & stats) const override { if (!realKey) return; realKey->mergeStats(stats); }
```

because you don't want to force the key to open when it hasn't been just because asking for stats. (for which there won't be any yet).